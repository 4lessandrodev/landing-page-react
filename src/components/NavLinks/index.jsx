import P from 'prop-types';
import Component from './style';
import MenuLink from '../MenuLink';

export const NavLinks = ({ links = [] }) => {
	return (<Component>
		{links.map((link, index) => (
			<MenuLink key={index} {...link}/>
		))}
	</Component>);
};

NavLinks.prototype = {
	links: P.arrayOf(
		P.shape({
			children: P.node.isRequired,
			href: P.string.isRequired,
			target: P.oneOf(['_blank', '_self'])
		})
	)
};

export default NavLinks;
