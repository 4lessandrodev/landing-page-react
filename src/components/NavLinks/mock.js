export default [
	{
		children: 'link 01',
		href: '#link01',
		target: '_blank'
	},
	{
		children: 'link 02',
		href: '#link02',
		target: '_blank'
	},
	{
		children: 'link 03',
		href: '#link03',
		target: '_blank'
	},
	{
		children: 'link 04',
		href: '#link04',
		target: '_blank'
	},
	{
		children: 'link 05',
		href: '#link05',
		target: '_blank'
	}
];
