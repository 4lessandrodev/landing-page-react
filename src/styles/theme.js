export const Theme = {
	colors: {
		primary: '#af2'
	},
	spacings: {
	}
};

export default Theme;
