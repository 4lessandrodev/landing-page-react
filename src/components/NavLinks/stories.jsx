import Component from './index';
import mock from './mock';

export default {
	title: 'NavLinks',
	component: Component,
	args: {
		links: mock
	}
};

export const Template = (args) => (<Component { ...args}/>);
