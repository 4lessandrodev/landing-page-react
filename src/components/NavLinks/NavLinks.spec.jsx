import { screen } from '@testing-library/dom';
import RenderWithTheme from '../../styles/render';
import NavLinks from './index';
import theme from '../../styles/theme';

import mock from './mock';

describe('<NavLinks />', () => {
	it('should be in document', () => {
		RenderWithTheme(<NavLinks links={ mock }/>);
		const element = screen.getAllByRole('link');
		expect(element).toHaveLength(5);
	});

	it('should not render links', () => {
		RenderWithTheme(<NavLinks links={ [] }/>);
		const element = screen.queryAllByRole('link');
		expect(element).toHaveLength(0);
	});

	it('should not render links to mobile', () => {
		RenderWithTheme(<NavLinks links={ mock }/>);
		const element = screen.getByText(/link 01/i).parentElement;
		expect(element).toHaveStyleRule('flex-flow', 'column wrap', {
			media: theme.media.lteMedium
		});
	});

	it('should match snapshot', () => {
		const { container } = RenderWithTheme(<NavLinks links={mock} />);
		expect(container.firstChild).toMatchSnapshot();
	});
});
