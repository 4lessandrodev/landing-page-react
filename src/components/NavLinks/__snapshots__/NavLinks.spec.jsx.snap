// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLinks /> should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #f96e46;
  display: block;
  position: relative;
  font-size: 1.6rem;
  padding: 0.8rem;
  font-weight: bold;
}

.c1::after {
  content: '';
  position: absolute;
  bottom: 0.4rem;
  left: 0%;
  width: 0;
  height: 0.3rem;
  background: #00e8fc;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c1:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    -webkit-align-content: center;
    -ms-flex-line-pack: center;
    align-content: center;
  }
}

<nav
  class="c0"
>
  <a
    class="c1"
    href="#link01"
    target="_blank"
  >
    link 01
  </a>
  <a
    class="c1"
    href="#link02"
    target="_blank"
  >
    link 02
  </a>
  <a
    class="c1"
    href="#link03"
    target="_blank"
  >
    link 03
  </a>
  <a
    class="c1"
    href="#link04"
    target="_blank"
  >
    link 04
  </a>
  <a
    class="c1"
    href="#link05"
    target="_blank"
  >
    link 05
  </a>
</nav>
`;
